### PR TITLE
Refactor Code to Use K8S Retries

### PR DESCRIPTION
**Pull request by briandowns**
## Proposed Changes ####

Refactors the application of PSPs, Role Bindings, Cluster Role's, and Cluster Role Bindings. This also adds unit tests for the changes. The unit tests increase coverage to 14.8%.

#### Linked Issues ####

rancher/rke2#207

### CODE REVIEW

**Review by brandond — APPROVED** [review]

**Review by brandond — CHANGES_REQUESTED** [review]

File: pkg/rke2/psp.go [hunk 1]
```diff
@@ -436,8 +436,10 @@ func deployClusterRoleBindingFromYaml(ctx context.Context, cs kubernetes.Interfa
 		},
 	); err != nil && apierrors.IsAlreadyExists(err) {
 		return retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-			_, err := cs.RbacV1().ClusterRoleBindings().Update(ctx, &clusterRoleBinding, metav1.UpdateOptions{})
-			return err
+			if _, err := cs.RbacV1().ClusterRoleBindings().Update(ctx, &clusterRoleBinding, metav1.UpdateOptions{}); err != nil {
```
[the remark this answers is not in the document]
Comment (briandowns) [reply pkg/rke2/psp.go:439]: Yeah, it was a remnant from testing.

File: pkg/rke2/psp.go [hunk 2]
```diff
@@ -389,116 +387,109 @@ func decodeYamlResource(data interface{}, yaml string) error {
...
-			return err
-		}
-		if _, err := cs.RbacV1().ClusterRoleBindings().Update(ctx, &clusterRoleBinding, metav1.UpdateOptions{}); err != nil {
+	// try to create the given cluster role binding. If it already exists, we
+	// fall through to attempting to update the existing cluster role binding.
+	if err := retry.OnError(retry.DefaultBackoff,
+		func(err error) bool {
+			return !apierrors.IsAlreadyExists(err)
+		}, func() error {
+			if _, err := cs.RbacV1().ClusterRoleBindings().Create(ctx, &clusterRoleBinding, metav1.CreateOptions{}); err != nil {
```
Comment (brandond) [inline pkg/rke2/psp.go:429]: One more bit of err handling simplification

**Review by brandond — CHANGES_REQUESTED** [review]
Other than this one bit, lgtm

**Review by brandond — APPROVED** [review]

**Review by galal-hussein — APPROVED** [review]
LGTM

**Review by brandond — APPROVED** [review]

**Review by dweomer — APPROVED** [review]

**Pull request merged**